tests/results/DiffTest.py

```diff
@@ -207,6 +209,305 @@ def test_from_string_arrays(self):
+        self.assertEqual(self.uut.modified, target)
+
+    def test_from_unified_diff_incomplete_hunks_multiple_deletions(self):
+        source = ['A', 'B', 'C', 'D', 'E', 'F', 'G']
+        target = ['A', 'C', 'D', 'E', 'F']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,1 @@',
+                ' A',
+                '-B',
+                '@@ -5,3 +4,2 @@',
+                ' E',
+                ' F',
+                '-G']
+        diff_string = '\n'.join(diff)
+        self.uut = Diff.from_unified_diff(diff_string, source)
+        self.assertEqual(self.uut.original, source)
+        self.assertEqual(self.uut.modified, target)
+
+    def test_from_unified_diff_incomplete_hunks_multiple_additions(self):
+        source = ['A', 'C', 'D', 'E', 'G']
+        target = ['A', 'B', 'C', 'D', 'E', 'F', 'G']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,1 +1,2 @@',
+                ' A',
+                '+B',
+                '@@ -4,2 +5,3 @@',
+                ' E',
+                '+F',
+                ' G']
+        diff_string = '\n'.join(diff)
+        self.uut = Diff.from_unified_diff(diff_string, source)
+        self.assertEqual(self.uut.original, source)
+        self.assertEqual(self.uut.modified, target)
+
+    def test_from_unified_diff_incomplete_hunks_multiple_modifications(self):
+        source = ['A', 'B', 'C', 'D', 'E', 'F', 'G']
+        target = ['A', 'B', 'Z', 'D', 'E', 'F', 'K']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,3 +1,3 @@',
+                ' A',
+                ' B',
+                '-C',
+                '+Z',
+                '@@ -6,2 +5,2 @@',
+                ' F',
+                '-G',
+                '+K']
+        diff_string = '\n'.join(diff)
+        self.uut = Diff.from_unified_diff(diff_string, source)
+        self.assertEqual(self.uut.original, source)
+        self.assertEqual(self.uut.modified, target)
+
+    def test_from_unified_diff_unmatched_line_to_delete(self):
+        source = ['first', 'second']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,2 @@',
+                '-line_to_be_deleted_is_not_same',
+                '+only_first_changed',
+                ' second']
+        diff_string = '\n'.join(diff)
+        with self.assertRaises(RuntimeError):
+            self.uut = Diff.from_unified_diff(diff_string, source)
+
+    def test_from_unified_diff_unmatched_context_line(self):
+        source = ['first', 'second']
+        diff = ['--- a/testfile',
+                '+++ b/testfile',
+                '@@ -1,2 +1,2 @@',
+                ' context_line_is_not_same',
+                ' second']
+        diff_string = '\n'.join(diff)
+        with self.assertRaises(RuntimeError):
+            self.uut = Diff.from_unified_diff(diff_string, source)
```

Comment:
this is something worth to test I think: Could you assert for the exception messages? As you print the lines, line-numbers etc it should be tested imo :+1: